website/blog/2024-11-25-announcing-lingui-5.0/index.md

```diff
@@ -0,0 +1,288 @@
+
+This change is breaking, meaning that some auto-generated message IDs may change. This could cause translations to lose their connection to the original message ID. To address this, you can use the translation memory capabilities of your translation management system (TMS) to restore translations for the updated message IDs.
+
+### New `useLingui` Macro
+
+In Lingui 5, we introduced a new [`useLingui`](/ref/macro#uselingui) macro that makes it easier to work with non-JSX messages within React components.
+
+Take the following code snippet as an example:
+
+```jsx
+import { t, msg } from "@lingui/macro";
+import { useLingui } from "@lingui/react";
+
+function MyComponent() {
+  const { i18n, _ } = useLingui();
+
+  const a = t(i18n)`Text`;
+}
+```
+
+With the new `useLingui` macro, you can simplify this code to:
+
+```jsx
+import { useLingui } from "@lingui/react/macro";
+
+function MyComponent() {
+  const { t } = useLingui();
+
+  const a = t`Text`;
+}
+```
+
+Under the hood, the `useLingui` macro is transpiled to the runtime [`useLingui`](/ref/react#uselingui) hook.
+
+### Standalone `babel-plugin-lingui-macro`
+
+There are now two ways to use Lingui macros with Babel in your project. You can choose between the `babel-macro-plugin` and the standalone `@lingui/babel-plugin-lingui-macro`.
+
+The new plugin offers faster transpiling and additional configuration options.
+
+We recommend replacing `babel-macro-plugin` with `@lingui/babel-plugin-lingui-macro` if you have direct access to your Babel configuration and don't use any other macros in your project.
+
+```diff title=".babelrc"
+{
+  "plugins": [
+-    "macros"
++    "@lingui/babel-plugin-lingui-macro"
+  ]
+}
+```
+
+### Print Placeholder Values for Better Translation Context
+
+Context is critical for translators (and AI) to provide accurate translations. This release introduces a new feature that prints placeholder values in PO comments.
+
+If the message contains unnamed placeholders such as `{0}`, Lingui will print their values in PO comments, giving translators and AI more context about what the placeholder is about.
+
+Consider the following example:
+
+```js
+t`Hello ${user.name} ${value}`;
+```
+
+With the new feature enabled, the message in the PO file will include the placeholder values in the comments:
+
+```diff title="en.po"
++ #. placeholder {0}: user.name
+msgid "Hello {0} {value}"
+```
+
+This gives the translators more context about the placeholders, which definitely helps in providing accurate translations.
+
+### Other Features
+
+#### Configurable Message Stripping via Babel Options
+
+By default, Lingui strips messages and comments from the production bundle to reduce the bundle size.
+
+It's now possible to enable this feature in development mode by setting the `stripMessageField` option in the Babel configuration:
```

Comment:
This actually might work vice-versa and disable stripping in production. Which I think has even more benefits for some users, because they will be able to manipulate original messages in runtime. 

We also need to note that this feature is not supported in SWC plugin (and probably i should sync it before we go live)